lib/reservation.rb

```diff
@@ -0,0 +1,19 @@
+
+module Hotel_System
+  class Reservation
+    attr_accessor :check_in, :check_out, :room, :rate_adjustor
+    def initialize(check_in, check_out)
+      raise ArgumentError.new("Check_in Date object, check_out Date object") unless check_in.class == Date && check_out.class == Date
+      raise ArgumentError.new("Cannot check in and out on same day") if check_in == check_out
+      raise ArgumentError.new("Time Warp") if (check_out <=> check_in) == -1
```

Comment:
There's some duplication for the Date logic/error handling between Reservation and Room. I wish there were a way you could pull this out to something reused, or at least placed in the same place!